internal/datacoord/task_scheduler.go

```diff
@@ -262,13 +278,31 @@ func (s *taskScheduler) process(taskID UniqueID) bool {
 			task.SetState(indexpb.JobState_JobStateRetry, "update meta building state failed")
 			return false
 		}
+		task.SetStartTime(time.Now())
+		queueingTime := task.GetStartTime().Sub(task.GetQueueTime())
+		if queueingTime > Params.DataCoordCfg.TaskSlowThreshold.GetAsDuration(time.Second) {
+			log.Warn("task queueing time is too long", zap.Int64("taskID", taskID),
+				zap.Int64("queueing time(ms)", queueingTime.Milliseconds()))
+		}
+		metrics.DataCoordTaskExecuteLatency.
+			WithLabelValues(task.GetTaskType(), metrics.Pending).Observe(float64(queueingTime.Milliseconds()))
+
 		log.Ctx(s.ctx).Info("update task meta state to InProgress success", zap.Int64("taskID", taskID),
 			zap.Int64("nodeID", nodeID))
 	case indexpb.JobState_JobStateFinished, indexpb.JobState_JobStateFailed:
 		if err := task.SetJobInfo(s.meta); err != nil {
 			log.Ctx(s.ctx).Warn("update task info failed", zap.Error(err))
 			return true
 		}
+		task.SetEndTime(time.Now())
+		runningTime := task.GetEndTime().Sub(task.GetStartTime())
+		if runningTime > Params.DataCoordCfg.TaskSlowThreshold.GetAsDuration(time.Second) {
+			log.Warn("task running time is too long", zap.Int64("taskID", taskID),
+				zap.Int64("running time(ms)", runningTime.Milliseconds()))
+		}
+		metrics.DataCoordTaskExecuteLatency.
```

Comment:
why we report this every time?
I believe we only need to report this once when index excution done?

Reply:
It will only be report once here. This is the task queue, and when a task is finished and the meta is successfully recorded, the task will be cleared from the queue.